scripts/automatic_tag_test.go

```diff
@@ -0,0 +1,199 @@
+	for _, test := range []struct {
+		desc               string
+		tag                *plumbing.Reference
+		latestTags         map[string]string
+		coreOnly           bool
+		expectedLatestTags map[string]string
+	}{
+		{
+			desc: "valid service latest version",
+			tag:  plumbing.NewReferenceFromStrings("refs/tags/services/foo/v0.2.0", ""),
+			latestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+			coreOnly: false,
+			expectedLatestTags: map[string]string{
+				"foo": "v0.2.0",
+				"bar": "v0.1.0",
+			},
+		},
+		{
+			desc: "valid service no update",
+			tag:  plumbing.NewReferenceFromStrings("refs/tags/services/foo/v0.1.0", ""),
+			latestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+			coreOnly: false,
+			expectedLatestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+		},
+		{
+			desc: "valid core latest version",
+			tag:  plumbing.NewReferenceFromStrings("refs/tags/core/v0.2.0", ""),
+			latestTags: map[string]string{
+				"foo":  "v0.1.0",
+				"bar":  "v0.1.0",
+				"core": "v0.1.0",
+			},
+			coreOnly: true,
+			expectedLatestTags: map[string]string{
+				"foo":  "v0.1.0",
+				"bar":  "v0.1.0",
+				"core": "v0.2.0",
+			},
+		},
+		{
+			desc: "valid core latest version",
+			tag:  plumbing.NewReferenceFromStrings("refs/tags/core/v0.2.0", ""),
+			latestTags: map[string]string{
+				"foo":  "v0.1.0",
+				"bar":  "v0.1.0",
+				"core": "v0.1.0",
+			},
+			coreOnly: true,
+			expectedLatestTags: map[string]string{
+				"foo":  "v0.1.0",
+				"bar":  "v0.1.0",
+				"core": "v0.2.0",
+			},
+		},
+		{
+			desc: "invalid service tag",
+			tag:  plumbing.NewReferenceFromStrings("refs/tags/foo/v0.2.0", ""),
+			latestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+			coreOnly: false,
+			expectedLatestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+		},
+		{
+			desc: "valid service tag but coreOnly",
+			tag:  plumbing.NewReferenceFromStrings("refs/tags/services/foo/v0.2.0", ""),
+			latestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+			coreOnly: true,
+			expectedLatestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+		},
+		{
+			desc: "dont update to pre-release",
+			tag:  plumbing.NewReferenceFromStrings("refs/tags/services/foo/v0.2.0-pre-release", ""),
+			latestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+			coreOnly: true,
+			expectedLatestTags: map[string]string{
+				"foo": "v0.1.0",
+				"bar": "v0.1.0",
+			},
+		},
+	} {
+		t.Run(test.desc, func(t *testing.T) {
+			updatedLatestTags := storeLatestTag(test.tag, test.latestTags, test.coreOnly)
+			if !reflect.DeepEqual(test.expectedLatestTags, updatedLatestTags) {
+				t.Fatalf("Updated tags are wrong, expected %v, got %v", test.expectedLatestTags, updatedLatestTags)
+			}
+		})
+	}
+}
+
+func TestComputeUpdatedTag(t *testing.T) {
```

Comment:
Also, could you add some test cases where the update is `minor` but the current patch version is not `0`?

E.g. `v0.1.2` -> `0.2.0`

Reply:
done